### ISSUE TITLE
get_text_responses admits a single string but iterates through the string

### ISSUE THREAD

**Issue by taniokay** [citadel-ai/langcheck#188]
Each eval client can take a single string, but it does not treat it as a single prompt, but a collection of a single character.
```Python
client.get_text_responses('Hello, can I ask a question?') # outputs 28 responses for each character
```

**Comment by taniokay** [comment]
@liwii Any specific reasons for `Iterable[str]` here?
https://github.com/citadel-ai/langcheck/blob/4ec7aec86e40478ca208ade0916969324c8c5353/src/langcheck/metrics/eval_clients/_base.py#L52

If not, we can explicitly admit a single prompt by: `prompts: list[str] | str', which is what `langcheck.augment` is doing now
https://github.com/citadel-ai/langcheck/blob/4ec7aec86e40478ca208ade0916969324c8c5353/src/langcheck/augment/en/_synonym.py#L50

**Comment by liwii** [comment]
Yeah it is really confusing... I faced the same issue once as well.

I'm not sure if there are specific reasons. I guess it carries over the types attached on some very old interface.

As long as replacing it with `list[str]` doesn't cause any errors, I'm fine with replacing that!
(I personally prefer limiting the type to `list[str]` only because `get_text_responses` is not meant to be called directly so often & `list[str]` makes the code cleaner, but the design is up to you!)
